actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -168,6 +168,112 @@ def method_missing(called, *args, **options, &block)
           end
       end
 
+      # :method: tag.attributes
+      #
+      # Transforms a Hash into HTML attributes, ready to be interpolated into
+      # ERB. Includes or omits boolean attributes based on their truthiness.
+      # Transforms keys nested within
+      # <tt>aria:</tt> or <tt>data:</tt> objects into <tt>aria-</tt> and <tt>data-</tt>
+      # prefixed attributes:
+      #
+      #   <input <%= tag.attributes(type: :text, aria: { label: "Search" }) %>>
+      #   # => <input type="text" aria-label="Search">
+      #
+      #   <button <%= tag.attributes id: "call-to-action", disabled: false, aria: { expanded: false } %> class="primary">Get Started!</button>
+      #   # => <button id="call-to-action" aria-expanded="false" class="primary">Get Started!</button>
+
+      # :method: tag.*
```

Comment:
I don't think we use this kind of label to methods. This is going to be the first. Can't we document all the html tags?

Reply:
This style of documentation is derived from entries like [ActiveStorage::Attached::Model.with_attached_*](https://edgeapi.rubyonrails.org/classes/ActiveStorage/Attached/Model.html#method-i-with_attached_-2A).

The splat is supposed to serve as a searchable value, since both `TagBuilder` (the class returned by `helpers.tag`) is marked with `:nodoc:`, and because the methods themselves rely on `method_missing` and aren't defined.

Reply:
The difference is that we don't have any documentation that is `object.method`. The `with_attached_*` doesn't have an object that is a prefix. The right way to document this would be to expose the tag builder in the public API and perhaps stop using `method_missing`. That is probably why it is so slow compared with the `tag("input")` method.

Reply:
If we have access to Nokogiri, we could loop over `Nokogiri::HTML::ElementDescription::DefaultDescriptions`:

```ruby
irb(main):006:0> Nokogiri::HTML::ElementDescription::DefaultDescriptions.map(&:first).take(10)
=> ["a", "abbr", "acronym", "address", "applet", "area", "b", "base", "basefont", "bdo"]
```

Reply:
We could, but is there any reason to not hard code the list? We already have a list of `HTML_VOID_ELEMENTS` and `SVG_SELF_CLOSING_ELEMENTS`.